Modules/IO/MeshGifti/src/itkGiftiMeshIO.cxx

```diff
@@ -168,45 +175,11 @@ GiftiMeshIO::ReadMeshInformation()
       }
       this->m_UpdatePoints = true;
 
-      switch (m_GiftiImage->darray[ii]->datatype)
+      this->m_PointComponentType = GetComponentTypeFromGifti(m_GiftiImage->darray[ii]->datatype);
+      if (GetNumberOfPixelComponentsFromGifti(m_GiftiImage->darray[ii]->datatype) > 1)
       {
-        case NIFTI_TYPE_INT8:
-          this->m_PointComponentType = IOComponentEnum::CHAR;
-          break;
-        case NIFTI_TYPE_UINT8:
-          this->m_PointComponentType = IOComponentEnum::UCHAR;
-          break;
-        case NIFTI_TYPE_INT16:
-          this->m_PointComponentType = IOComponentEnum::SHORT;
-          break;
-        case NIFTI_TYPE_UINT16:
-          this->m_PointComponentType = IOComponentEnum::USHORT;
-          break;
-        case NIFTI_TYPE_INT32:
-          this->m_PointComponentType = IOComponentEnum::INT;
-          break;
-        case NIFTI_TYPE_UINT32:
-          this->m_PointComponentType = IOComponentEnum::UINT;
-          break;
-        case NIFTI_TYPE_INT64:
-          this->m_PointComponentType = IOComponentEnum::LONGLONG;
-          break;
-        case NIFTI_TYPE_UINT64:
-          this->m_PointComponentType = IOComponentEnum::ULONGLONG;
-          break;
-        case NIFTI_TYPE_FLOAT32:
-          this->m_PointComponentType = IOComponentEnum::FLOAT;
-          break;
-        case NIFTI_TYPE_FLOAT64:
-          this->m_PointComponentType = IOComponentEnum::DOUBLE;
-          break;
-        case NIFTI_TYPE_FLOAT128:
-          this->m_PointComponentType = IOComponentEnum::LDOUBLE;
-          break;
-        default:
-          itkExceptionMacro(<< "Unknown point component type");
+        itkExceptionMacro(<< "Invalid datatype in data array " << ii << " detected.");
```

Comment:
A more descriptive message here might be helpful. Something like: "Pointset ii is expected to contain only scalars".